hpx/util/spinlock.hpp

```diff
@@ -28,9 +28,15 @@ struct spinlock
 
   public:
     spinlock(char const* /*desc*/ = nullptr)
+#if BOOST_VERSION >= 106200
       : m{BOOST_DETAIL_SPINLOCK_INIT}
+#endif
     {
         HPX_ITT_SYNC_CREATE(this, "util::spinlock", "");
+#if BOOST_VERSION < 106200
+        boost::detail::spinlock l = BOOST_DETAIL_SPINLOCK_INIT;
+        m = l;
+#endif
```

Comment:
A simple m(BOOST_DETAIL_SPINLOCK_INIT) should do it just fine, and works for older boost versions as well.